### PR TITLE
Support inline variable assignment in Include tag

### PR DESCRIPTION
**Pull request by hishamco**
This is a part of issue #30

### CODE REVIEW

**Comment by hishamco** [comment]
@sebastienros  is passing the assignment statements into the include is a good idea, or there is way to add them to the context, something similar to `TemplateContext.GlobalScope`

File: Fluid/DefaultFluidParser.cs [hunk 1]
```diff
@@ -893,7 +901,28 @@ public static FilterArgument BuildFilterArgument(ParseTreeNode node)
...
+        {
+            var identifier = tag.ChildNodes[0].Token.ValueString;
+            var value = BuildTermExpression(tag.ChildNodes[1]);
+
+            return new AssignStatement(identifier, value);
+        }
+
         #endregion
+
+        private static void Traverse(ParseTreeNode tag, ref IList<AssignStatement> statements)
```
Comment (sebastienros) [inline Fluid/DefaultFluidParser.cs]: Why `ref`?

Comment (sebastienros) [reply Fluid/DefaultFluidParser.cs]: What is recursive in include assignments?

Comment (hishamco) [reply Fluid/DefaultFluidParser.cs]: > Why ref?

The `ref` needed to keep track the `AssignStatement`s during the traversing, I could make as global field instead and remove from here .. I think this a better choice
> What is recursive in include assignments?

The recursive needed to visit all the node in the assignment tree

File: Fluid/Ast/IncludeStatement.cs [hunk 2]
```diff
@@ -45,6 +48,14 @@ public IncludeStatement(Expression path)
                 throw new FileNotFoundException(relativePath);
             }
 
+            if (AssignStatements != null)
+            {
+                foreach (var assignStatement in AssignStatements)
```
Comment (sebastienros) [inline Fluid/Ast/IncludeStatement.cs]: I don't think these values should be visible after the include statement is executed. Probably necessary to call `context.EnterScope` and `content.LeaveScope` after.

File: Fluid/Ast/IncludeStatement.cs [hunk 3]
```diff
@@ -50,10 +50,14 @@ public IncludeStatement(Expression path, IList<AssignStatement> assignStatements
 
             if (AssignStatements != null)
             {
+                context.EnterChildScope();
```
Comment (sebastienros) [inline Fluid/Ast/IncludeStatement.cs]: You need to grab the result and pass it to `assignStatement.WriteToAsync`

Comment (hishamco) [reply Fluid/Ast/IncludeStatement.cs]: I didn't get your point, what I understand is to make the assignments executed within the scope, similar to https://github.com/sebastienros/fluid/blob/dev/Fluid/Ast/IncludeStatement.cs#L51-L65

Correct me if I'm wrong

Comment (sebastienros) [reply Fluid/Ast/IncludeStatement.cs]: I was wrong ;)

Comment (sebastienros) [reply Fluid/Ast/IncludeStatement.cs]: I just didn't remember what I had done :blush:

Comment (hishamco) [reply Fluid/Ast/IncludeStatement.cs]: 😄

**Comment by hishamco** [comment]
LGTM, when need to rebase to merge both of the PRs

thanks @sebastienros

**Pull request merged**